src/shred_collector/shred.zig

```diff
@@ -47,33 +47,62 @@ pub const Shred = union(ShredType) {
         };
     }
 
-    pub fn fromPayload(allocator: Allocator, payload: []const u8) !Self {
-        const variant = layout.getShredVariant(payload) orelse return error.InvalidShredVariant;
+    pub fn fromPayload(allocator: Allocator, payload_: []const u8) !Self {
+        const variant = layout.getShredVariant(payload_) orelse return error.InvalidShredVariant;
         return switch (variant.shred_type) {
-            .Code => .{ .Code = .{ .fields = try CodingShred.Fields.fromPayload(allocator, payload) } },
-            .Data => .{ .Data = .{ .fields = try DataShred.Fields.fromPayload(allocator, payload) } },
+            .Code => .{ .Code = .{ .fields = try CodingShred.Fields.fromPayload(allocator, payload_) } },
```

Comment:
im saying its not clear that `fromPayload` would have sanitize logic from its name - one reason is because its similar to `bincode.fromSlice`, which has no sanitize logic

a method name which describes that it will read and sanitize would be better imo eg, `DeserializeAndSanitize` or even `DeserializeAndValidate` (not a big fan of `sanitize` - was more so just following agave's approach) or something along those lines

if you find the function names in bincode confusing, we can always change them to something more clear so we can get consistency across the codebase - not in this PR, but create a ticket with some other ideas and we can come up with something better

Reply:
for the sake of time - ill resolve this, create a ticket, and we can come back to it later